aws/transit_gateway_test.go

```diff
@@ -0,0 +1,307 @@
+	sleepFor := 180 * time.Second
+	sleepWithMessage(sleepFor, sleepMessage)
+
+	return *result.TransitGatewayRouteTable
+}
+
+func TestGetAllTransitGatewayRouteTableInstances(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	require.NoError(t, err)
+
+	tgwRouteTableName := "cloud-nuke-test-" + util.UniqueID()
+	tgwRouteTable := createTestTransitGatewayRouteTable(t, session, tgwRouteTableName)
+
+	defer nukeAllTransitGatewayRouteTables(session, []*string{tgwRouteTable.TransitGatewayRouteTableId})
+	defer nukeAllTransitGatewayInstances(session, []*string{tgwRouteTable.TransitGatewayId})
+
+	ids, err := getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour*-1))
+	require.NoError(t, err)
+	assert.NotContains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+
+	ids, err = getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour))
+	require.NoError(t, err)
+	assert.Contains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+}
+
+func TestNukeTransitGatewayRouteTable(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&awsgo.Config{
+		Region: awsgo.String(region)},
+	)
+	require.NoError(t, err)
+
+	svc := ec2.New(session)
+
+	tgwRouteTableName := "cloud-nuke-test-" + util.UniqueID()
+	tgwRouteTable := createTestTransitGatewayRouteTable(t, session, tgwRouteTableName)
+	defer nukeAllTransitGatewayInstances(session, []*string{tgwRouteTable.TransitGatewayId})
+
+	_, err = svc.DescribeTransitGatewayRouteTables(&ec2.DescribeTransitGatewayRouteTablesInput{
+		TransitGatewayRouteTableIds: []*string{
+			tgwRouteTable.TransitGatewayRouteTableId,
+		},
+	})
+	require.NoError(t, err)
+
+	err = nukeAllTransitGatewayRouteTables(session, []*string{tgwRouteTable.TransitGatewayRouteTableId})
+	require.NoError(t, err)
+
+	ids, err := getAllTransitGatewayRouteTables(session, region, time.Now().Add(1*time.Hour))
+	require.NoError(t, err)
+	assert.NotContains(t, awsgo.StringValueSlice(ids), awsgo.StringValue(tgwRouteTable.TransitGatewayRouteTableId))
+}
+
+func createTestTransitGatewayVpcAttachment(t *testing.T, session *session.Session, name string) ec2.TransitGatewayVpcAttachment {
+	svc := ec2.New(session)
+
+	transitGateway := createTestTransitGateway(t, session, name)
+
+	input := &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   awsgo.String("isDefault"),
+				Values: []*string{awsgo.String("true")},
+			},
+		},
+	}
+
+	vpcs, err := svc.DescribeVpcs(input)
+	if err != nil {
+		logging.Logger.Error(t, "TransitGatewayVpcAttachment test depends on default VPC availability")
+		assert.Fail(t, errors.WithStackTrace(err).Error())
+	}
+	assert.NoError(t, err)
+
+	vpc := vpcs.Vpcs[0]
```

Comment:
I think this should be updated to use `require` and a length check. Like this:

```
vpcs, err := svc.DescribeVpcs(input)
require.NoError(t, err)
require.Len(t, vpcs.Vpcs, 1)
vpc := vpcs.Vpcs[0]
```

Reply:
Sounds good! implemented!